src/test/test_bitcoin.cpp

```diff
@@ -45,7 +58,7 @@ BasicTestingSetup::BasicTestingSetup(const std::string& chainName)
     // TODO: fix the code to support SegWit blocks.
     gArgs.ForceSetArg("-vbparams", strprintf("segwit:0:%d", (int64_t)Consensus::BIP9Deployment::NO_TIMEOUT));
     SelectParams(chainName);
-    noui_connect();
+    noui_connect_noop_handlers();
```

Comment:
Thinking more about this, could this only be disabled for specific test cases? Otherwise writing tests will get a lot harder when errors and warnings are no longer printed.

Reply:
Good point. Pushed updated version. No longer trying to suppress any output. The spamming problem was solved by running `noui_connect()` only once :-)

Reply:
> Otherwise writing tests will get a lot harder when errors and warnings are no longer printed.

Yes, thanks, good point—in my experience errors being mysteriously swallowed is awful for being able to diagnose problems when they happen.